File: SegundaPregunta.py

suma = lambda x, y: x + y
resta = lambda x, y: x - y
multiplicacion = lambda x, y: x * y
division = lambda x, y: x / y


def calculadora(x, y, op):
    if op == '+':
        print(suma(x, y))
    if op == '-':
        print(resta(x, y))
    if op == '*':
        print(multiplicacion(x, y))
    if op == '/':
        print(division(x, y)) 
        
calculadora(14, 7, '+')

File: SegundaPregunta.scala
val suma = (x:Int, y:Int) => x + y
val resta = (x:Int, y:Int) => x - y
val multiplica = (x:Int, y:Int) => x * y
val divide = (x:Int, y:Int) => x / y

def calculadora(x:Int, y:Int, operacion:String):Int= {
	operacion match {
	case "suma" => suma(x,y)
	case "resta" => resta(x,y)
	case "multiplica" => multiplica(x,y)
	case "divide" => divide(x,y)
	case "" => -1
	}
}

// se llama de la siguiente forma
val resultado = calculadora(6, 7, "suma")